pkg/app/carbonapi/http_handlers.go

```diff
@@ -704,8 +705,17 @@ func (app *App) renderWriteBody(results []*types.MetricData, form renderForm, r
 
 	switch form.format {
 	case jsonFormat:
-		if maxDataPoints, _ := strconv.Atoi(r.FormValue("maxDataPoints")); maxDataPoints != 0 {
-			results = types.ConsolidateJSON(maxDataPoints, results)
+		// let's limit maxDataPoints to math.MaxInt32, i.e. 2147483647
+		umaxDataPoints64, err2 := strconv.ParseInt(r.FormValue("maxDataPoints"), 10, 32)
+		if err2 != nil {
+			if umaxDataPoints64 >= 0 && umaxDataPoints64 <= math.MaxInt32 {
+				maxDataPoints := int(umaxDataPoints64)
+				results = types.ConsolidateJSON(maxDataPoints, results)
+			} else {
+				return nil, fmt.Errorf("maxDataPoints %d should be >=0 and <=%d", umaxDataPoints64, math.MaxInt32)
```

Comment:
in idiomatic go error conditions are checked before happy-path


https://go.dev/doc/effective_go#control-structures
> This is an example of a common situation where code must guard against a sequence of error conditions. The code reads well if the successful flow of control runs down the page, eliminating error cases as they arise. Since error cases tend to end in return statements, the resulting code needs no else statements.